plugins/destination/duckdb/client/types.go

```diff
@@ -13,36 +14,49 @@ type listLike interface {
 }
 
 func transformSchemaForWriting(sc *arrow.Schema) *arrow.Schema {
-	fields := sc.Fields()
+	md := arrow.MetadataFrom(sc.Metadata().ToMap())
+	return arrow.NewSchema(transformFieldsForWriting(sc.Fields()), &md)
+}
+
+func transformFieldsForWriting(fields []arrow.Field) []arrow.Field {
 	for i := range fields {
 		fields[i].Type = transformTypeForWriting(fields[i].Type)
 	}
-	md := sc.Metadata()
-	return arrow.NewSchema(fields, &md)
+	return fields
 }
 
 func transformTypeForWriting(dt arrow.DataType) arrow.DataType {
 	switch dt := dt.(type) {
+	case *arrow.StructType:
+		return arrow.StructOf(transformFieldsForWriting(dt.Fields())...)
+	case *arrow.MapType:
+		return arrow.MapOf(transformTypeForWriting(dt.KeyType()), transformTypeForWriting(dt.ItemType()))
 	case listLike:
 		return arrow.ListOf(transformTypeForWriting(dt.Elem()))
-	case *arrow.MapType:
-		return arrow.ListOf(transformTypeForWriting(dt.ValueType()))
-	}
-
-	switch dt := duckDBToArrow(arrowToDuckDB(dt)).(type) {
 	case *types.UUIDType, *types.JSONType:
 		return arrow.BinaryTypes.String
 	default:
-		return dt
+		return duckDBToArrow(arrowToDuckDB(dt))
```

Comment:
why?

Reply:
it was switch by `duckDBToArrow(arrowToDuckDB(dt))`